From 7a4f8aaee6429082ceb514785a8bb33d9ca65d55 Mon Sep 17 00:00:00 2001
From: Mia Lin <mimi05633@gmail.com>
Date: Fri, 25 Aug 2023 10:48:58 +0800
Subject: [PATCH 1/1] PLDM: Requester: Add markFree method

Mark a PLDM instance id as unused, for the input MCTP EID.

Code base on
  commit 402dc6a5beee58ee1e6514a425c542a0d879954c
  Author: Michal Orzel <michalx.orzel@intel.com>
  Date:   Thu Jul 27 15:33:12 2023 +0200

      Change authority D-Bus names

Signed-off-by: Mia Lin <mimi05633@gmail.com>
---
 .../PLDM/Requester.interface.yaml                | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/yaml/xyz/openbmc_project/PLDM/Requester.interface.yaml b/yaml/xyz/openbmc_project/PLDM/Requester.interface.yaml
index 928f7b87..182c0e96 100644
--- a/yaml/xyz/openbmc_project/PLDM/Requester.interface.yaml
+++ b/yaml/xyz/openbmc_project/PLDM/Requester.interface.yaml
@@ -49,3 +49,19 @@ methods:
                 PLDM instance id.
       errors:
           - xyz.openbmc_project.Common.Error.TooManyResources
+
+    - name: markFree
+      description: >
+          Mark a PLDM instance id as unused, for the input MCTP EID.
+      parameters:
+          - name: eid
+            type: byte
+            description: >
+                The MCTP endpoint, specified by 'eid' (endpoint id), for which
+                the new PLDM instance id needs to be generated.
+          - name: instanceid
+            type: byte
+            description: >
+                PLDM instance id.
+      errors:
+         - xyz.openbmc_project.Common.Error.InvalidArgument
-- 
2.17.1

